src/Compilers/CSharp/Test/Syntax/Parsing/TypeArgumentListParsingTests.cs

```diff
@@ -1875,5 +2267,215 @@ void M()
+                                                            N(SyntaxKind.IdentifierName);
+                                                            {
+                                                                N(SyntaxKind.IdentifierToken, "U");
+                                                            }
+                                                        }
+                                                        N(SyntaxKind.LessThanToken);
+                                                        N(SyntaxKind.IdentifierName);
+                                                        {
+                                                            N(SyntaxKind.IdentifierToken, "T");
+                                                        }
+                                                    }
+                                                    N(SyntaxKind.LessThanToken);
+                                                    N(SyntaxKind.UnsignedRightShiftExpression);
+                                                    {
+                                                        N(SyntaxKind.IdentifierName);
+                                                        {
+                                                            N(SyntaxKind.IdentifierToken, "S");
+                                                        }
+                                                        N(SyntaxKind.GreaterThanGreaterThanGreaterThanToken);
+                                                        N(SyntaxKind.IdentifierName);
+                                                        {
+                                                            N(SyntaxKind.IdentifierToken, "ProjectChange");
+                                                        }
+                                                    }
+                                                }
+                                                N(SyntaxKind.EqualsToken);
+                                                N(SyntaxKind.IdentifierName);
+                                                {
+                                                    N(SyntaxKind.IdentifierToken, "projectChange");
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                                N(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void TestGenericArgWithGreaterThan_03()
+        {
+            UsingTree(@"
+class C
+{
+    void M()
+    {
+        var added = ImmutableDictionary<T<S>>>
```

Comment:
> Would it be useful to have a test with a tuple type next to the ```>>>``` as well?

I am not aware any specific reason why this could be useful, other than the fact that any test potentially has some value. Also, I think there is no test like that for ```>>```, so I assume there isn't anything special there.